text/0000-argument-lifetimes.md

```diff
@@ -0,0 +1,492 @@
+
+Thus, where today you would write:
+
+```rust
+fn elided(&self) -> &str;
+fn two_args<'a, 'b>(arg1: &'a Foo, arg2: &'b Bar) -> &'b Baz;
+
+impl<'a> MyStruct<'a> {
+    fn foo(&self) -> &'a str;
+    fn bar<'b>(&self, arg: &'b str) -> &'b str;
+}
+
+fn take_fn<'a>(x: &'a u32, y: for<'b> fn(&'a u32, &'b u32, &'b u32))
+```
+
+tomorrow you would write:
+
+```rust
+fn elided(&self) -> &str;
+fn two_args(arg1: &Foo, arg2: &'b Bar) -> &'b Baz;
+
+impl MyStruct<'A> {
+    fn foo(&self) -> &'A str;
+    fn bar(&self, arg: &'b str) -> &'b str;
+}
+
+fn take_fn(x: &'a u32, y: fn(&'a u32, &'b u32, &'b u32));
+```
+
+## The wildcard lifetime
+
+When referring to a type (other than `&`/`&mut`) that requires lifetime
+arguments, it is deprecated to leave off those parameters.
+
+Instead, you can write a `_` for the parameters, rather than giving a lifetime
+name, which will have identical behavior to leaving them off today.
+
+Thus, where today you would write:
+
+```rust
+fn foo(&self) -> Ref<SomeType>
+fn iter(&self) -> Iter<T>
+```
+
+tomorrow you would write:
+
+```rust
+fn foo(&self) -> Ref<_, SomeType>
+fn iter(&self) -> Iter<_, T>
+```
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The deprecations here involve some amount of churn (largely in the form of
+deleting lifetimes from `<>` blocks, but also sometimes changing their
+case). Users exercise a lot of control over when they address that churn and can
+do so incrementally. Moreover, we can and should consider addressing this with a
+`rustfix`, which should be easy and highly reliable.
+
+The fact that lifetime parameters are not bound in an out-of-band way is
+somewhat unusual and might be confusing---but then, so are lifetime parameters!
+Putting the bindings out of band buys us very little, as argued in the next
+section.
+
+Introducing a case distinction for lifetime names is a bit clunky, and of course
+not all spoken languages *have* a case distinction (but for those, another
+convention could be applied).
+
+Requiring a `_` rather than being able to leave off lifetimes altogether may be
+a slight decrease in ergonomics in some cases.
+
+Cases where you could write `fn foo<'a, 'b: 'a>(...)` now need the `'b: 'a` to
+be given in a `where` clause, which might be slightly more verbose. These are
+relatively rare, though, due to our type well-formedness rule.
+
+Otherwise, it's a bit hard to see drawbacks here: nothings is made more explicit
+or harder to determine, since the binding structure continues to be completely
+unambiguous; ergonomics and, arguably, learnability both improve. And
+signatures become less noisy and easier to read.
+
+# Rationale and Alternatives
+[alternatives]: #alternatives
+
+## Core rationale
+
+The key insight of the proposed design is that out-of-band bindings for lifetime
+parameters is buying us very little today:
+
+- For free functions, it's completely unnecessary; the only lifetime "in scope"
+  is `'static`, so everything else *must* be a parameter.
```

Comment:
Something like `fn pick_any_lifetime<'a>() -> &'a str { "hello" }` is a perfectly cromulent free function that uses a lifetime not coupled to any parameter. Would this continue to compile?